### ISSUE TITLE
External Library Non-existent path file was accidentally deleted.

### ISSUE THREAD

**Issue by Gloryandel** [immich-app/immich#10039]
### The bug

Removing offline files accidentally deletes files in external libraries that were once under the add path.

![image](https://github.com/immich-app/immich/assets/51497852/66d2dd68-5e87-4716-bf02-d1767932ca74)

### The OS that Immich Server is running on

Arch

### Version of Immich Server

v1.105.1

### Version of Immich Mobile App

v1.105.1

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
name: immich
services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: ['start.sh', 'immich']
    environment:
      DB_USERNAME: ${DB_USERNAME}
      DB_PASSWORD: ${DB_PASSWORD}
      DB_DATABASE_NAME: ${DB_DATABASE_NAME}
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
      - /srv/pool/data/angel/files/Pictures:/mnt/Pictures
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: unless-stopped

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/hardware-transcoding
      file: hwaccel.transcoding.yaml
      service: quicksync # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    command: ['start.sh', 'microservices']
    environment:
      DB_USERNAME: ${DB_USERNAME}
      DB_PASSWORD: ${DB_PASSWORD}
      DB_DATABASE_NAME: ${DB_DATABASE_NAME}
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
      - /srv/pool/data/angel/files/Pictures:/mnt/Pictures
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: unless-stopped
```
```


### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=/srv/pool/appdata/immich/library
# The location where your database files are stored
DB_DATA_LOCATION=/srv/pool/appdata/immich/postgres

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=password

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=postgres
DB_DATABASE_NAME=immich
```

### Reproduction steps

```bash
1. Create a new external library in immich web, add path  `/mnt/Pictures` and scan for library files.
2. When the file scanning is complete, remove the path that was just added in the external library. The file is offline.Note that the results are normal up to this point.
3. Remove Offline Files, A file under the path `/mnt/Pictures` that does not exist in the external library was accidentally deleted.It is naive to think that merely deleting thumbnails, database information.
![image_28](https://github.com/immich-app/immich/assets/51497852/9667655f-4f12-4878-83d6-271cdea179b9)


```

### Relevant log output

```shell
normal
```

**Comment by Ricostyle21** [comment]
Just writing to let you know that this issue is still a problem in 1.22.1 with external libraries, excluding already added and scanned folders and then pressing "Permanently delete" from the trash. The file gets deleted in the newly excluded folder on disk and is irretrievable. When setting the mount to read-only (as @iamrizzle did), and "permanently delete"-ing from the ui like before, I get the same error message unlinking the file.

**Comment by Gloryandel** [comment]
> Just writing to let you know that this issue is still a problem in 1.22.1 with external libraries, excluding already added and scanned folders and then pressing "Permanently delete" from the trash. The file gets deleted in the newly excluded folder on disk and is irretrievable. When setting the mount to read-only (as @iamrizzle did), and "permanently delete"-ing from the ui like before, I get the same error message unlinking the file. Is this still unintended behaviour @jrasm91 ?

Your intention seems strange. You have already moved it to the Recycle Bin and clicked Permanent Delete, which clearly deleted it. Why is the file still deleted? I think it should be deleted. Whether it is in the ignore path or not, you have indicated that you no longer need it.

**Comment by Ricostyle21** [comment]
> Your intention seems strange. You have already moved it to the Recycle Bin and clicked Permanent Delete, which clearly deleted it. Why is the file still deleted? I think it should be deleted. Whether it is in the ignore path or not, you have indicated that you no longer need it.

My bad, I should have pointed out: the photos that were scanned once and then excluded were moved to the trash by Immich itself. Since there are no obvious other ways (to me, but maybe I'm missing something) to clear the generated thumbnails, videos and database entries, I figured I needed to empty the trash to get rid of these. Unfortunately that had some unintended consequences.

On top of that, when I removed the entire external library, it removed most (but not all; 91 out of 23000 thumbnails remained) of the thumbnails, and none of the transcoded videos. Might be just my instance, but it was still strange.
